CMakeLists.txt

```diff
@@ -277,109 +236,15 @@ if (ARMA_EXTRA_DEBUG)
   add_definitions(-DARMA_EXTRA_DEBUG)
 endif()
 
-# Now, find the libraries we need to compile against.  Several variables can be
-# set to manually specify the directory in which each of these libraries
-# resides.
-#   ARMADILLO_LIBRARY - location of libarmadillo.so / armadillo.lib
-#   ARMADILLO_INCLUDE_DIR - directory containing <armadillo>
-#   ARMADILLO_INCLUDE_DIRS - directories necessary for Armadillo includes
-#   CEREAL_INCLUDE_DIR - include directory for cereal
-#   ENSMALLEN_INCLUDE_DIR - include directory for ensmallen
-#   STB_IMAGE_INCLUDE_DIR - include directory for STB image library
-#   MATHJAX_ROOT - root of MathJax installation
-
-# Download and compile OpenBLAS if we are cross compiling mlpack for a specific
-# architecture. The function takes the version of OpenBLAS as variable.
 if (CMAKE_CROSSCOMPILING)
-  search_openblas(0.3.28)
-endif()
-
-if (NOT DOWNLOAD_DEPENDENCIES)
-  find_package(Armadillo "${ARMADILLO_VERSION}" REQUIRED)
+  fetch_mlpack_and_crosscompile()
```

Comment:
If you wanted, you could change the macros you have a little bit and make `fetch_mlpack()` check the `CMAKE_CROSSCOMPILING` value internally.  Then all you would need is `fetch_mlpack()`.  Maybe there is some complexity I overlooked, I'm not sure.

Reply:
I like the idea and it is easy to do, but what about the user perspective? I do not want someone to have some doubt if this will happen or not. In essence, I don't want someone to not be sure how to do it. Let me think about it